rfcs/rfc-0000-pronouns.md

```diff
@@ -0,0 +1,70 @@
+- Feature Name: Pronouns
+- Start Date: (16/06/2023)
+- RFC PR: None
+- Tracking Issue: None
+- Status: draft
+
+# Summary
+
+This change creates a pronoun field for revolt's backend.
+
+
+# Motivation
+
+Pronouns have been requested by the community and are high up on the RFC list.
+
+# Guide-level explanation
+
+Currently users display their pronouns in their Profile content (bio) but this provices a cleaner solution to the problem. The aim of this change is to ensure that pronouns for a user can be easily seen and clearly displayed seperately to a users bio.
+
+# Reference-level explanation
+
+This change will modify:
+- core/models/users
+- core/db/util/bridge
+- quark/models/users
+- core/db/models/users/ops/mongo
+  And possibly more
+
+For the backend, pronoun data will be stored as such.
+```rust
+    // User's displayed pronouns
+        #[cfg_attr(feature = "validator", validate(length(min = 1, max = 5)))]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub pronouns: Option<Vec<String>>,
+```
+
+Having pronouns on Revolt will be optional and this data will be editable from the Profile section of user settings on Revolt's main client. One of the major changes this could bring is pronoun specific language for bots interacting with users.
+
+#### Input
+```text
+?/ whois info:roles @ToastXC
+```
+#### Response
+```text
+She has no roles!
+```
+
+# Drawbacks
+
+Mocking users for displaying pronouns on their profile and creating fake pronouns as a joke is very common and seen by some as hateful, There must be a system built into the pronoun system to avoid this. That would take some time and may not be worth the effort.
```

Comment:
I'm not sure how per-server pronouns would work. I can see it being abused. I think it should be a case of pronouns being on a profile or not. Anyone determined to out somebody or otherwise view their pronouns would likely just be able to click the user's account and view it there. While I don't see much of a point being able to change them per server I do see a case for being able to enable/disable their visibility per server.

Reply:
Oh yeah in retrospect those two are mutually exclusive, if you didn't have a visibility toggle it would be nice to override it per server, otherwise I guess it doesn't apply?

Reply:
Though, maybe you have a case where, you want to show your pronouns on all servers, but you're only out to one of those servers. In that case you may not want to update your pronouns globally.

Reply:
That is a valid use case I hadn't considered. I think that could be very beneficial. This would also require some changes to how the 'Server Identity' popup works.

Reply:
That'll likely be done anyways in the new client.

Reply:
assuming pronouns become part of the server identity menu, it might be nice to have a dedicated page in the options menu to view and configure every custom server identity you currently have set up. should help make updating pronouns across multiple servers a lot easier

Reply:
I like the idea of having toggles for where pronouns should be visible. However, DMs may need more customization like a choice between visibility for:
- Everyone: allows anyone you dm to see your pronouns (For when DMs are possible between non-friends)
- Friends: only friends can see your pronouns in DMs
- Select: choose select people/groups that would be able to view your pronouns (For the current system where you can only DM friends. This would prevent some stranger who wants to talk one-on-one from learning your pronouns, if that is an issue. It could also be used once you can DM non-friends where someone you are friends with yet aren't out to won't be able to see your pronouns)

Reply:
isn't it already impossible to dm someone without adding them as a friend first?

Reply:
I believe so, however iirc there were plans to make it a toggle to allow dms with non-friends? Unless that is only being considered and not absolutely planned.

Reply:
> if you hide pronouns, you may start to get accidentally misgendered by strangers. this defeats the point of displaying pronouns.

I just want to stress that, there still may be individuals in a situation where they are out to some people but not others, so they may not want to display their pronouns to certain groups of people. This is a situation where you'd expect to be misgendered, but it's something that you're okay with because you haven't communicated any change to those individuals.

Reply:
I assume if someone is hiding pronouns they expect the possibility of being misgendered, but would rather not out themselves to combat it and just ignore it. Or they can correct the people they trust if those people can't see the pronouns do to hiding toggles. Either way, it seems better to have an option for people to have the choice of hiding pronouns, rather than having to select an incorrect pair of pronouns to those they're not out to. Not to mention, I feel most people who would make someone feel unsafe to come out to would be more against showing *any* pronouns anyway.

Reply:
If pronouns are going to become part of server identity (for the purpose of protection and avoiding harassment) should users be prompted to confirm their server identity before being added to the server? Otherwise there will be a period of time when their default pronouns will be visible to other users in the server.

Reply:
^ that'd actually be a rly good idea i think

Reply:
hey sorry I'm late to this thread.
One idea i had was; only allowing mutual friends to be able to see pronouns if toggled by the user.

Reply:
While per-server pronouns are an option (as they can be set in the server identity field) I'm not sure if this is the best solution; as generally speaking users have the same pronouns on every server unless I'm missing something

Reply:
I believe that Revolt should be safe from bigots and groups who would target people based on pronouns, and while it is possible to engineer a bot or otherwise to target groups with gender neutral pronouns I think the solution should be moderation, not changes to per-server identity or privacy.

Reply:
> I believe that Revolt should be safe from bigots and groups who would target people based on pronouns, and while it is possible to engineer a bot or otherwise to target groups with gender neutral pronouns I think the solution should be moderation, not changes to per-server identity or privacy.

As a moderator of Revolt Lounge, I have absolutely no idea why you would think Revolt would be safe from bigots and groups of bigots? I genuinely don't understand this comment.

The point on bots is probably a bit unrealistic, or rather, a server that decides to add such a bot is probably run by someone who would be targeting people anyway so it doesn't matter too much.

> While per-server pronouns are an option (as they can be set in the server identity field) I'm not sure if this is the best solution; as generally speaking users have the same pronouns on every server unless I'm missing something

I think per-server pronouns are important for anyone who wants to engage with friends or communities but has yet to come out and therefore would prefer to use certain pronouns in certain places. The assumption is that there would be a default set of pronouns that you select and then you have to manually override them for each server to get different ones - rather than set them up per server.

My point on prompting before you join would be more like "Do you want to use your default settings for this server? Otherwise, you can override them now." Not, "Enter your details from scratch to join this server."

Speaking of friends, how should pronouns be displayed to friends or in DMs? Does there need to be per-friend pronouns? Could there be 2 groups of friends, that could see different sets of pronouns when speaking to you? Should these settings override any per-server settings? Is this getting a bit complicated? ... Probably, but it's all good stuff.

Friend groups; Normal and Trusted. Normal sees one set of pronouns. Trusted sees a different set. Pronoun settings for friends overrides any per-server pronouns to ensure you don't accidentally out yourself to a friend you haven't come out to yet if they join your lgbt server.

If you'd like to take it one further... (going a bit wild here). Any amount of friend groups (trusted, family, work, minecraft, guild, etc. custom groups) where each group can have a different identity much like per-server identities. And an option for these identities to override per-server identities - or, at least, pronouns and bios.

----

So, went a bit off course there but the main point is that I think it's important to protect people who have yet to come out but still want to explore their identity in safe spaces/servers.

Reply:
While Revolt is not perfect for moderation, generally the team is very fast to respond to bigots and is fairly responsible. We have very little tolerance to bigotry and trolling on the lounge server but unfortunately due to its size and publicity it attracts some bad eggs. I say that Revolt is a safe place for LGBTQIA people because the core team is committed to fighting and removing harmful content with many new projects under development to automate moderation as well as increase the number of active moderators in different timezones.

Reply:
The server override idea is being considered as it could be a part of the server identity field without much modification, although questions of privacy are still being discussed.

Reply:
> I believe that Revolt should be safe from bigots and groups who would target people based on pronouns, and while it is possible to engineer a bot or otherwise to target groups with gender neutral pronouns I think the solution should be moderation, not changes to per-server identity or privacy.

i'm sorry, but i really, really do not like this kind of thinking. 

while i do believe platform moderators are perfectly capable of handling issues like this, preventing problems is *always* going to be better than solving them *after* they've already happened. this goes for so many issues in society as a whole, and it's incredibly frustrating to see people continue to stick with "we'll just deal with it if/when it happens".

Reply:
I am tempted to agree with the actual proposal here too - I can think of plenty of instances where you might not be out to certain groups of people (**especially** IRL friends/family) or might want to hide your identity to lower suspicion/avoid mistreatment - if nothing else, a per-server toggle on whether to show them would be a good start imo